test/format/ops.c

```diff
@@ -659,7 +659,7 @@ prepare_transaction(TINFO *tinfo)
 #define OP_FAILED(notfound_ok)                                                               \
     do {                                                                                     \
         positioned = false;                                                                  \
-        if (intxn && (ret == WT_CACHE_FULL || ret == WT_ROLLBACK))                           \
+        if (intxn && (ret == WT_CACHE_FULL || ret == WT_ROLLBACK || ret == WT_RESTART))      \
```

Comment:
Would test/format handle it correctly if we returned `WT_RESTART` here instead? That way we avoid nuking the entire transaction and can just retry the single operation.

Reply:
We discussed this offline. @jiechenbo did you end up making a separate ticket for enhancing test/format to understand `WT_RESTART` and not re-do the entire txn?

Reply:
We never return WT_RESTART from WiredTiger at the moment, it's reserved for internal use. The documentation in our public API confirms that:

```
/*! Restart the operation (internal). */
#define WT_RESTART      (-31805)
```

We restrict the error codes users of WiredTiger need to be aware of aggressively, to make it easier to program against WiredTiger. The return value we have for "retry" is `WT_PREPARE_CONFLICT`, which doesn't really match this usage.

I'd be inclined to return a WT_ROLLBACK with a good rollback reason describing why.

Reply:
I have now moved it to returning WT_ROLLBACK, let me know if the message looks good to you. [bcb782c](https://github.com/wiredtiger/wiredtiger/pull/10311/commits/bcb782c2a9519c6debb93acf0e6fa5d29e294ba3)

Reply:
I'm happy with it, but will leave the thread open until Alex G is also happy.